### DISCUSSION TITLE
LottieConfiguration shared instance in Swift 6

### DISCUSSION THREAD

**Discussion by prengifo** [airbnb/lottie-ios#2484] [General]
## Which Version of Lottie are you using?

Lottie 4.5.0

## Problem

While enabling complete concurrency checks for Swift 5 in Xcode 16, we get a warning that accessing the shared instance from `LottieConfiguration` is not concurrency-safe. This should be tagged correctly with some concurrency strategy to avoid the warning.

```
LottieConfiguration.shared.renderingEngine = .mainThread
```

![image](https://github.com/user-attachments/assets/d6f0dcb8-fc22-4660-a8d2-42b2b907b320)

## Expected Behavior 
No warning should be raised by the compiler

## Actual Behavior
A warning is raised in Swift 5 with complete concurrency check, while an error will be raised in Swift 6.

**Comment by calda** [comment]
Static mutable globals like this are never concurrency-safe. 

Making Lottie compatible with Swift 6 mode and strict concurrency checking will require probably require breaking API changes.
